### PR TITLE
Migrate mmap-related syscall handlers (mman.h) to rust

### PR DESCRIPTION
**Pull request by stevenengler**
Hopefully I didn't mess anything up. Could be cleaned up a bit, but tried to stay similar to the existing code.

I also fixed what I think is a bug, where the fd we open in the plugin is not closed in a specific code path.

### CODE REVIEW

File: src/main/host/syscall/handler/mman.rs
Comment (robgjansen) [inline src/main/host/syscall/handler/mman.rs:288]: Would it be easy enough to at least detect when this is happening and log a warning?

File: src/main/host/syscall/handler/mman.rs
Comment (robgjansen) [inline src/main/host/syscall/handler/mman.rs:255]: Do the tests still pass if we remove the extra native syscall try?

**Review by robgjansen — APPROVED** [review]

File: src/main/host/syscall/handler/mman.rs
Comment (stevenengler) [reply src/main/host/syscall/handler/mman.rs:288]: I added a check for the few cases that shadow supports.

File: src/main/host/syscall/handler/mman.rs
Comment (stevenengler) [reply src/main/host/syscall/handler/mman.rs:255]: I removed this code and added an assertion that the result is not `SyscallError::Native`. I'll wait to get your approval before I merge.

**Review by robgjansen — APPROVED** [review]

**Pull request merged**
